Monika-After-Story/MonikaModDev
# mas_songs_database

#3054 

Songs are now a part of their own event database

# Additions:
- Two new delegates, a random delegate and a pool delegate
- Random delegate shows songs one time, and then they're unlocked in the pool delegate
- Pool delegate holds songs which have been seen before
- Songs (random ones, not pooled) work with `aff_range`. (Not pool because if you've seen them before, it kinda makes sense to be able to ask Monika to sing it again), though pool is Normal+
- Song Types, (`TYPE_LONG` and `TYPE_SHORT`). Note, that we should never use type long for random songs, but pool only. As such there should be some other form of unlocking them (as getting stuck in long random topics would probably kinda suck) (These are filtered from the random pick)
- update scripts

## New methods:
- `checkRandSongDelegate()`
  - Unlocks the random song delegate if it were derandomed (because no more random songs)
  - This allows re-rand when more songs are added automatically
- `getUnlockedSongs()`
  - gets all unlocked songs
- `getRandomSongs()`
  - gets all songs that are random and that we're in the aff_range for
- `hasUnlockedSongs()`
  - checks if we've unlocked any songs
- `hasRandomSongs()`
  - checks if there are any more random songs available

# Testing:
- Check shown counts of all songs before updating the game files
- Update files and run update script label (`v0_10_2`)
- Verify the shown counts of the new evs reflect that of the old ones and that ones which you have seen are not random, and are unlocked
- Restore a persistent, and change some aff ranges on the random songs. Verify those you're out of the range for don't show up in the list when you call `mas_songs.getRandomSongs()`
